### ISSUE TITLE
MySQL Bug

### ISSUE THREAD

**Issue by safepage** [martinjw/dbschemareader#85]
Since v6.10 of MySQL Connector/NET, the DbProviderFactory method CreateDataAdapter returns null. For some reason this is seen as Severity S3 (Non-critical) and has still not been fixed in the latest v8.0.

For details please see:
https://bugs.mysql.com/bug.php?id=88660

Whist I appreciate this is not an issue with dbschemareader, one of the suggested work-arounds is to use: https://github.com/mysql-net/MySqlConnector/

But this means supporting a different MySql driver in dbschemareader (i.e. not MySql.Data.MySqlClient).

Do you have any thoughts on this?

Thanks

**Comment by martinjw** [comment]
I haven't tried MySqlConnector, but I use the Devart MySql provider ($$) as well as MySql.Data and they work the same. It should work...

DbSchemaReader is multi-targeted, there's several .net versions (3.5-4.5) and netstandard 1.5 and 2.0. The netstandard versions of dbSchemaReader don't use DbProviderFactory, so you'd be able to use that with MySql even with their borked factory.  If the main project is Core, you'll get the netstandard version. If you use 4.6.1+ you can use a netstandard library to get the nuget package (there are some annoying nuget/build issues, but workarounds exist).

**Comment by safepage** [comment]
I use .Net 4.5, so I base my project on DatabaseSchemaReader2015.csproj. I modified this by adding Defined Constants for COREFX;NETSTANDARD2_0 and removing all the files as per the netstandard1.5 target...and it now works OK!

One other issue, the current MySQL 8.0.12 has another bug (https://bugs.mysql.com/bug.php?id=90690) which means it throws when getting the Constraints.

To work around this, I changed DatabaseSchemaReader\ProviderSchemaReaders\Databases\MySql\Constraints.cs Line 47 from:

`cons2.constraint_name = refs.unique_constraint_name AND`

To:

`cons2.constraint_name COLLATE utf8_unicode_ci = refs.unique_constraint_name COLLATE utf8_unicode_ci AND`

This is marked as fixed in the next release MySQL 8.0.13, but currently means DbSchemaReader doesn't work.

Thanks
